activesupport/test/core_ext/duration_test.rb

```diff
@@ -763,6 +766,71 @@ def test_duration_symmetry
     assert_equal expected_time, time + ActiveSupport::Duration.build(-1)
   end
 
+  def test_time_ext_deprecated
+    deprecated_numeric_methods = %i[second seconds minute minutes hour hours day days week weeks fortnight fortnights]
+    deprecated_integer_methods = %i[month months year years]
+    not_deprecated = %i[in_milliseconds]
+
+    (deprecated_numeric_methods + deprecated_integer_methods).each do |method|
+      assert_deprecated do
+        1.year.public_send(method)
+      end
+    end
+
+    not_deprecated.each do |method|
+      assert_not_deprecated do
+        1.year.public_send(method)
+      end
+    end
+  end
+
+  def test_missing_methods_deprecated
+    methods = %i[
+      byte bytes day days exabyte exabytes fortnight fortnights gigabyte
+      gigabytes hour hours kilobyte kilobytes megabyte megabytes minute
+      minutes month months petabyte petabytes second seconds terabyte terabytes
+      week weeks year years
+
+      & ** << >> [] ^ abs2 allbits? angle anybits? arg bit_length ceil chr conj
+      conjugate denominator digits downto even? finite? floor gcd gcdlcm i imag
+      imaginary infinite? lcm magnitude multiple_of? next nobits? numerator odd?
+      ord ordinal ordinalize phase polar pow pred rationalize real real? rect
+      rectangular round size step succ times to_bn to_c truncate upto | ~
+    ]
+
+    methods.each do |method|
+      assert_deprecated do
+        1.year.public_send(method) rescue ArgumentError
```

Comment:
That `rescue` syntax isn't doing what it appears

Reply:
Hmm, I had added it because some of the deprecated methods take an argument and this seemed to quiet those errors. Am I missing something about how that works? (I figured the `assert_deprecated` is ensuring that the deprecation is getting printed whether or not the delegated method call succeeds)

Reply:
Thank you for the explanation!